infrastructure/builder/src/utils/repo.js

```diff
@@ -57,17 +57,24 @@ exports.writeRepoJSON = async (filename, data) => {
 
 /**
  * Modify a file in-place in the current working copy, calling `await modifier(contents)`.
+ * This function is "sequentialized" so that concurrent modifications do not interfere.
  *
  * The file is assumed to be utf-8.
  */
-exports.modifyRepoFile = async (filename, modifier) => {
+const modifyRepoFile = async (filename, modifier) => {
   const contents = await readFile(
     path.join(REPO_ROOT, filename),
     {encoding: 'utf8'});
   const modified = await modifier(contents);
   await writeFile(filename, modified, {encoding: 'utf8'});
 };
 
+let modifyRepoPromise = Promise.resolve();
+exports.modifyRepoFile = (filename, modifier) => {
+  modifyRepoPromise = modifyRepoPromise.catch(() => {}).then(() => modifyRepoFile(filename, modifier));
```

Comment:
what is the purpose of `modifyRepoPromise`?

Reply:
It's serializing the promises -- there is a comment up above the function describing it.  I ran into a bug where two things were calling `modifyRepoFile('.taskcluster.yml', ..)` and overwriting each other.  But this is kind of a mess.  Maybe I can find a library for it?  We do this a few other places, too.  I've made a followup bug for that (https://bugzilla.mozilla.org/show_bug.cgi?id=1561668)